nntrainer/tensor/blas_interface.h

```diff
@@ -152,22 +152,56 @@ void sgemv(CBLAS_ORDER order, CBLAS_TRANSPOSE TransA, const unsigned int M,
            const unsigned int lda, const _FP16 *X, const int incX,
            const float beta, _FP16 *Y, const int incY);
 /**
- * @brief     elementwise vector multiplication : Z = X ⊙ Y
+ * @brief     elementwise vector multiplication : Z = X ⊙ alpha * Y +
+ * beta * Z
  * @param[in] N  length of the vector
- * @param[in] X __fp16 * for Vector X
- * @param[in] Y __fp16 * for Vector Y
- * @param[in] Z __fp16 * for Vector Z
- */
-void ewvm(const unsigned int N, const _FP16 *X, const _FP16 *Y, _FP16 *Z);
+ * @param[in] X _FP16 * for Vector X
+ * @param[in] Y _FP16 * for Vector Y
+ * @param[in] Z _FP16 * for Vector Z
+ * @param[in] alpha scalar multiplier for input
```

Comment:
quick question. it seems like scalars are added only for Y and Z. wouldn't there be cases where X also needs a scalar?

Reply:
also, why is scalar added in the first place?

Reply:
1. Good point. Ususally, strict BLAS functions only get X and Y, not Z. So it would go like : $X = X * \alpha (op) Y * \beta$. But current function usage in the NNTrainer requires all $X, Y, Z$  . 
2. from tensor operation, we differentiate whether to use std::transform or ele_* by stride and alpha / beta. For more broader use of this function. scalar multiplier is necessary.

Reply:
thanks for clarification :)